weather_app.py

```diff
@@ -0,0 +1,50 @@
+import requests
+
+# Data Source Class
+class HourlyTemperature:
+    def __init__(self, temperatures: list):
+        self.temperatures = temperatures
+
+# Service Class
+class WeatherService:
+    def __init__(self, api_url: str):
+        self.api_url = api_url
+
+    def get_hourly_temperature(self, latitude: float, longitude: float) -> HourlyTemperature:
+        params = {
+            'latitude': latitude,
+            'longitude': longitude,
+            'hourly': 'temperature_2m'
+        }
+        response = requests.get(self.api_url, params=params)
+        response.raise_for_status()
+        data = response.json()
```

Comment:
here you could consider more error handling (erroneous response, bad data types etc.)